contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -252,7 +252,7 @@ pub struct SubmitBid<'info> {
 
     pub relayer_signer: Signer<'info>,
 
-    /// CHECK: this is the permission_key
+    /// CHECK: this is the permission_key. This is included as an account to save transaction size via caching in case of repeated use; account contestation is not a concern because this account should never be written to.
```

Comment:
> because this account should never be written to

This account can be mutable in the same transaction via the other programs using it, so this part may be a bit misleading (although this program never changes it).
Maybe we can say something like:
... to  save transaction size. Usually the permission_key can be described as the account storing the opportunity information, so we just reuse that account to save 32 bytes instead of putting this in instruction arguments.

Reply:
you're right that other programs can set it to mutable, but that's generally true for all accounts (except programs). The comment here expresses that this account's data _should never_ be mutated; I'll make that clearer